include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -1542,6 +1570,7 @@ namespace internal
                       nullptr,
                       gradients_quad,
                       nullptr,
+                      nullptr,
```

Comment:
I think it would make more sense to pass the argument on here, 
```suggestion
                      hessians_quad,
                      nullptr,
```
the you can remove the assert from here and rely on the `FEEvaluationImplCollocation` class.